qcore/microtime.py

```diff
@@ -110,27 +107,62 @@ def __init__(self, offset):
         self.offset = int(offset)
 
     def __enter__(self):
-        global _time_offset
-        _time_offset += self.offset
+        global _offset_utime
+        _offset_utime += self.offset
 
     def __exit__(self, typ, value, traceback):
-        global _time_offset
-        _time_offset -= self.offset
+        global _offset_utime
+        _offset_utime -= self.offset
 
 
 def utime():
     """Gets current time in microseconds from the epoch time w/applied offset."""
-    return _time_offset + int(_time() * 1000000)
+    return _offset_utime + int(_time_in_seconds() * SECOND)
 
 
 def true_utime():
     """Gets current time in microseconds from the epoch time."""
-    return int(_time() * 1000000)
+    return int(_time_in_seconds() * SECOND)
+
+
+# ===================================================
+# Conversions to/from PY Date-Time
+# ===================================================
+
+
+def utime_as_datetime(utime, *, drop_subseconds=False, tz=timezone.utc):
+    """Get Python datetime instance for the given microseconds time.
+
+    This time refers to an absolute moment, given as microseconds from Unix Epoch.
+
+    """
+    seconds = int(utime / SECOND) if drop_subseconds else utime / SECOND
```

Comment:
Can also use floor division instead of `int()`.

Reply:
CY doesn't like `//`, apparently! :(